### ISSUE TITLE
Typing band colors in reverse causes issues

### ISSUE THREAD

**Issue by benjamin051000** [chaseruskin/ohm#1]
Cool project! I typed some band colors in backwards, just for fun.
```bash
ohm on  trunk is 📦 v1.0.0 via 🦀 v1.69.0 
❯ ./target/release/ohm brown black red gold
Identification: -[┃┃┃  ┃ ]-
Resistance: 1000.0 Ω ± 5.0% (min: 950.0 Ω, max: 1050.0 Ω)

ohm on  trunk is 📦 v1.0.0 via 🦀 v1.69.0 
❯ ./target/release/ohm gold red black brown
error: band Gold is not allowed as a digit
```
Technically, it's the same resistor. While I understand that intuitively I should know that gold is a tolerance value and therefore should come last, beginners may not realize this. Would it be possible to check the reverse before erroring out? Perhaps printing a message like "Hint: You got it backwards" for educational purposes?

Great little project! Way easier than looking up yet another website, so I'm definitely adding this to my toolbox. 🤠

**Comment by chaseruskin** [comment]
This is a great idea. So when an error is encountered during resistor band decoding, the decoding function is again called but with the inputted bands in reverse order. If this decoding was correct, an additional hint is displayed in the error message:
```
error: band Gold is not allowed as a digit (hint: try reversing the band order)
```
If this reversed decoding was also an error, the user just entered an illegal resistor color code combination and the original error will be displayed.

Thanks for trying it out and liking the project!

**Comment by chaseruskin** [comment]
Fixed with commit c4860bbc1f4bf9711aa4c8116a9f61ee49cebe95.
